src/mig/modules/migoval/migoval.go

```diff
@@ -0,0 +1,227 @@
+			r.Results.Errors = append(r.Results.Errors, fmt.Sprintf("%v", e))
+			r.Results.Success = false
+			endCounters()
+			r.Results.Statistics = stats
+			err, _ := json.Marshal(r.Results)
+			resStr = string(err)
+			return
+		}
+	}()
+
+	startCounters()
+
+	// Read module parameters from stdin
+	err := modules.ReadInputParameters(in, &r.Parameters)
+	if err != nil {
+		panic(err)
+	}
+
+	err = r.ValidateParameters()
+	if err != nil {
+		panic(err)
+	}
+
+	ovallib.Init()
+	ovallib.SetMaxChecks(r.Parameters.MaxConcurrentEval)
+
+	e := &elements{}
+
+	if len(r.Parameters.PkgMatch.Matches) > 0 {
+		oresp := ovallib.PackageQuery(r.Parameters.PkgMatch.Matches)
+		for _, x := range oresp {
+			npi := &PkgInfo{PkgName: x.Name, PkgVersion: x.Version, PkgType: x.PkgType}
+			e.Matches = append(e.Matches, *npi)
+		}
+
+		r.Results.Success = true
+		if len(e.Matches) > 0 {
+			r.Results.FoundAnything = true
+		}
+		r.Results.Elements = e
+		endCounters()
+		r.Results.Statistics = stats
+		buf, err := json.Marshal(r.Results)
+		if err != nil {
+			panic(err)
+		}
+		resStr = string(buf)
+		return
+	} else if r.Parameters.OvalDef != "" {
+		b := bytes.NewBufferString(r.Parameters.OvalDef)
+		decoder := base64.NewDecoder(base64.StdEncoding, b)
+		gz, err := gzip.NewReader(decoder)
+		if err != nil {
+			panic(err)
+		}
+		ovalbuf, err := ioutil.ReadAll(gz)
+		if err != nil {
+			panic(err)
+		}
+
+		od, err := ovallib.ParseBuffer(string(ovalbuf))
+		if err != nil {
+			panic(err)
+		}
+		ovalresults, err := ovallib.Execute(od)
+		if err != nil {
+			panic(err)
+		}
+		for _, x := range ovalresults {
+			if !r.Parameters.IncludeFalse {
+				if x.Status == ovallib.RESULT_FALSE {
+					continue
+				}
+			}
+			nmor := &MOResult{}
+			nmor.Title = x.Title
+			nmor.Status = x.StatusString()
+			nmor.ID = x.ID
+			e.OvalResults = append(e.OvalResults, *nmor)
+		}
+
+		r.Results.Success = true
+		if len(e.OvalResults) > 0 {
+			r.Results.FoundAnything = true
+		}
+		r.Results.Elements = e
+		endCounters()
+		r.Results.Statistics = stats
+		buf, err := json.Marshal(r.Results)
+		if err != nil {
+			panic(err)
+		}
+		resStr = string(buf)
+		return
+	}
+
+	panic("no function specified")
+	return
+}
+
+func (r Runner) ValidateParameters() (err error) {
+	if r.Parameters.MaxConcurrentEval <= 0 || r.Parameters.MaxConcurrentEval > 10 {
+		return fmt.Errorf("concurrent evaluation must be between > 0 and <= 10")
+	}
+	return
+}
+
+func (r Runner) PrintResults(result modules.Result, foundOnly bool) (prints []string, err error) {
+	var elem elements
+
+	err = result.GetElements(&elem)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, x := range elem.Matches {
+		resStr := fmt.Sprintf("pkgmatch name=%v version=%v type=%v", x.PkgName, x.PkgVersion, x.PkgType)
+		prints = append(prints, resStr)
+	}
+
+	for _, x := range elem.OvalResults {
+		resStr := fmt.Sprintf("ovalresult id=%v title=\"%v\" outcome=%v", x.ID, x.Title, x.Status)
+		prints = append(prints, resStr)
+	}
+
```

Comment:
Updated to print errors; I didn't add statistics yet as this isn't really complete, I am going to be adding some stats around data sizes and so on, so will add statistics once this is there.